File: Habitica/src/main/java/com/habitrpg/android/habitica/api/MaintenanceApiService.java
package com.habitrpg.android.habitica.api;

import com.habitrpg.common.habitica.models.responses.MaintenanceResponse;

import io.reactivex.rxjava3.core.Flowable;
import retrofit2.http.GET;

public interface MaintenanceApiService {

    @GET("maintenance-android.json")
    Flowable<MaintenanceResponse> getMaintenanceStatus();

    @GET("deprecation-android.json")
    Flowable<MaintenanceResponse> getDepricationStatus();

}

File: Habitica/src/main/java/com/habitrpg/android/habitica/apiService/MaintenanceApiService.kt
package com.habitrpg.android.habitica.apiService

import com.habitrpg.shared.habitica.models.responses.MaintenanceResponse
import retrofit2.http.GET

interface MaintenanceApiService {
    @GET("maintenance-android.json")
    suspend fun getMaintenanceStatus(): MaintenanceResponse?

    @GET("deprecation-android.json")
    suspend fun getDeprecationStatus(): MaintenanceResponse?
}
